wasm/demo/src/index.js

```diff
@@ -1,11 +1,156 @@
 import './style.css';
-import 'codemirror/lib/codemirror.css';
 import 'xterm/lib/xterm.css';
+import CodeMirror from 'codemirror';
+import 'codemirror/mode/python/python';
+import 'codemirror/addon/comment/comment';
+import 'codemirror/lib/codemirror.css';
+import { Terminal } from 'xterm';
+import LocalEchoController from 'local-echo';
+
+let rp;
 
-// A dependency graph that contains any wasm must all be imported
-// asynchronously. This `index.js` file does the single async import, so
-// that no one else needs to worry about it again.
-import('./main.js').catch(e => {
-    console.error('Error importing `main.js`:', e);
-    document.getElementById('error').textContent = e;
+// A dependency graph that contains any wasm must be imported asynchronously.
+import('rustpython')
+    .then(rustpy => {
+        rp = rustpy;
+        // so people can play around with it
+        window.rp = rustpy;
+        onReady();
+    })
+    .catch(e => {
+        console.error('Error importing `rustpython`:', e);
+        document.getElementById('error').textContent = e;
+    });
+
+const editor = CodeMirror.fromTextArea(document.getElementById('code'), {
+    extraKeys: {
+        'Ctrl-Enter': runCodeFromTextarea,
+        'Cmd-Enter': runCodeFromTextarea,
+        'Shift-Tab': 'indentLess',
+        'Ctrl-/': 'toggleComment',
+        'Cmd-/': 'toggleComment',
+        Tab: editor => {
+            var spaces = Array(editor.getOption('indentUnit') + 1).join(' ');
+            editor.replaceSelection(spaces);
+        }
+    },
+    lineNumbers: true,
+    mode: 'text/x-python',
+    indentUnit: 4,
+    autofocus: true
 });
+
+const consoleElement = document.getElementById('console');
+const errorElement = document.getElementById('error');
+
+function runCodeFromTextarea() {
+    // Clean the console and errors
+    consoleElement.value = '';
+    errorElement.textContent = '';
+
+    const code = editor.getValue();
+    try {
+        rp.pyExec(code, {
+            stdout: output => {
+                const shouldScroll =
+                    consoleElement.scrollHeight - consoleElement.scrollTop ===
+                    consoleElement.clientHeight;
+                consoleElement.value += output;
+                if (shouldScroll) {
+                    consoleElement.scrollTop = consoleElement.scrollHeight;
+                }
+            }
+        });
+    } catch (err) {
+        if (err instanceof WebAssembly.RuntimeError) {
+            err = window.__RUSTPYTHON_ERROR || err;
+        }
+        errorElement.textContent = err;
+        console.error(err);
+    }
+}
+
+const snippets = document.getElementById('snippets');
+
+function updateSnippet() {
+    const selected = snippets.value;
+
+    // the require here creates a webpack context; it's fine to use it
+    // dynamically.
+    // https://webpack.js.org/guides/dependency-management/
+    const snippet = require(`raw-loader!../snippets/${selected}.py`);
+
+    editor.setValue(snippet);
```

Comment:
When I run this locally I get the following error when trying to switch snippet:
```
Uncaught TypeError: string.split is not a function
    at splitLinesAuto (codemirror.js?56b3:680)
    at Doc.splitLines (codemirror.js?56b3:6452)
    at Doc.eval (codemirror.js?56b3:6111)
    at Doc.eval [as setValue] (codemirror.js?56b3:3941)
    at CodeMirror.eval [as setValue] (codemirror.js?56b3:9682)
    at HTMLSelectElement.updateSnippet (index.js?b635:83)
```
Changing the following works for me:
```js
editor.setValue(snippet.default);
```

Is this something only I get?

Reply:
`raw-loader` was probably updated, but I have a local `yarn.lock` that I have in my git exclude file. I'll look into it.

Reply:
It should be fixed now.

Reply:
Should be fixed now.